src/TextFieldCatalog.js

```diff
@@ -7,56 +7,84 @@ import ReactGA from 'react-ga';
 import './styles/TextFieldCatalog.scss';
 import {gtagCategory, gtagTextFieldAction} from './constants';
 
+const TEXT_FIELD_MAX_LENGTH = 18;
+
 const TextField = (props) => {
   const {
     textFieldId, outlined, textarea,
-    dense, leading, trailing, helperText, className, onClick
+    dense, leading, trailing, helperText, characterCounter, className, onClick,
+    noLabel, placeholder
   } = props;
   const classes = classnames('mdc-text-field', 'text-field', className, {
     'mdc-text-field--outlined': outlined,
     'mdc-text-field--textarea': textarea,
     'mdc-text-field--dense': dense,
     'mdc-text-field--with-leading-icon': leading,
     'mdc-text-field--with-trailing-icon': trailing,
+    'mdc-text-field--no-label': noLabel,
   });
+  const hasHelperLine = helperText || characterCounter;
+  const maxLengthValue = characterCounter ? TEXT_FIELD_MAX_LENGTH : null;
 
   return (
     <div className='text-field-container'>
       <div className={classes} ref={textFieldEl => textFieldEl && new MDCTextField(textFieldEl)}>
+        {characterCounter && textarea ? <CharacterCounter /> : null}
         {leading && <i className='material-icons mdc-text-field__icon'>event</i>}
-        {textarea ? <TextArea textFieldId={textFieldId} onClick={onClick} /> : <Input textFieldId={textFieldId} onClick={onClick}/>}
-        {outlined || textarea ? null : <Label textFieldId={textFieldId} dense={dense}/>}
+        {textarea ?
+          <TextArea textFieldId={textFieldId} onClick={onClick} maxLength={maxLengthValue} placeholder={placeholder} noLabel={noLabel} /> :
+          <Input textFieldId={textFieldId} onClick={onClick} maxLength={maxLengthValue} placeholder={placeholder} noLabel={noLabel} />}
+        {outlined || textarea || noLabel ? null : <Label textFieldId={textFieldId} dense={dense}/>}
         {trailing && <i className='material-icons mdc-text-field__icon'>delete</i>}
-        {outlined || textarea ? <Outline textFieldId={textFieldId}/> : <div className='mdc-line-ripple'></div>}
+        {outlined || textarea ? <Outline noLabel={noLabel} textFieldId={textFieldId}/> : <div className='mdc-line-ripple'></div>}
       </div>
-      {helperText ? <HelperText /> : null}
+      { hasHelperLine ?
+        <div className='mdc-text-field-helper-line'>
+          {helperText ? <HelperText /> : null}
+          {characterCounter && !textarea ? <CharacterCounter /> : null}
+        </div>
+        : null
+      }
     </div>
   );
 }
 
-const FullWidthTextField = ({textarea, textFieldId, helperText}) => {
-  const classes = classnames('mdc-text-field', 'text-field', 'mdc-text-field--fullwidth', {
+const FullWidthTextField = ({textarea, textFieldId, helperText, characterCounter}) => {
+  // TODO: Remove --no-label class name in next release.
```

Comment:
Just checking, but I assume this has to do with the bug we fixed after 0.44.0?

Reply:
That's correct. It was fixed in https://github.com/material-components/material-components-web/pull/4385 (after v0.44.0 release)